[bug] update_policy=legacy breaks lock create

### Describe the bug

When the update_policy is set to legacy, the `conan lock create` command fails when creating a new lockfile from an input lockfile.

### How to reproduce it

```
conan lock create conanfile.py --lockfile-out base.lock --build='*' --update
conan lock create conanfile.py --lockfile base.lock --lockfile-out full.lock --build='*' --update
```

The first command works, the second fails with this error:
```
ERROR: Traceback (most recent call last):
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/graph/python_requires.py", line 101, in _resolve_py_requires
    py_require = self._cached_py_requires[resolved_ref]
                 ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^
KeyError: euler_conanfile/2.4.1@sick/release#29f906e8ee36b386c93164756e6677b8%1736951024.778

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/asaljo/.local/lib/python3.11/site-packages/conan/cli/cli.py", line 307, in main
    cli.run(args)
  File "/home/asaljo/.local/lib/python3.11/site-packages/conan/cli/cli.py", line 192, in run
    command.run(self._conan_api, args[0][1:])
  File "/home/asaljo/.local/lib/python3.11/site-packages/conan/cli/command.py", line 195, in run
    sub.run(conan_api, parser, *args)
  File "/home/asaljo/.local/lib/python3.11/site-packages/conan/cli/command.py", line 213, in run
    info = self._method(conan_api, parent_parser, self._parser, *args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conan/cli/commands/lock.py", line 43, in lock_create
    graph = conan_api.graph.load_graph_consumer(path, args.name, args.version,
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conan/api/subapi/graph.py", line 143, in load_graph_consumer
    root_node = self._load_root_consumer_conanfile(path, profile_host, profile_build,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conan/api/subapi/graph.py", line 25, in _load_root_consumer_conanfile
    conanfile = app.loader.load_consumer(path,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/loader.py", line 162, in load_consumer
    conanfile = self.load_named(conanfile_path, name, version, user, channel, graph_lock,
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/loader.py", line 104, in load_named
    conanfile, _ = self.load_basic_module(conanfile_path, graph_lock, remotes=remotes,
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/loader.py", line 66, in load_basic_module
    self._pyreq_loader.load_py_requires(conanfile, self, graph_lock, remotes,
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/graph/python_requires.py", line 81, in load_py_requires
    py_requires = self._resolve_py_requires(py_requires_refs, graph_lock, loader, remotes,
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/graph/python_requires.py", line 103, in _resolve_py_requires
    pyreq_conanfile = self._load_pyreq_conanfile(loader, graph_lock, resolved_ref,
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/graph/python_requires.py", line 124, in _load_pyreq_conanfile
    recipe = self._proxy.get_recipe(ref, remotes, update, check_update)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/graph/proxy.py", line 27, in get_recipe
    resolved = self._get_recipe(ref, remotes, update, check_update)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/graph/proxy.py", line 59, in _get_recipe
    remote, remote_ref = self._find_newest_recipe_in_remotes(reference, remotes,
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/graph/proxy.py", line 114, in _find_newest_recipe_in_remotes
    refs = self._remote_manager.get_recipe_revisions_references(reference, remote)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asaljo/.local/lib/python3.11/site-packages/conans/client/remote_manager.py", line 219, in get_recipe_revisions_references
    assert ref.revision is None, "get_recipe_revisions_references of a reference with revision"
AssertionError: get_recipe_revisions_references of a reference with revision

ERROR: get_recipe_revisions_references of a reference with revision
```

When I disable the policy, the build works fine. @memsharded ping

--- a/conans/client/graph/proxy.py
+++ b/conans/client/graph/proxy.py
@@ -106,7 +106,7 @@ class ConanProxy:
                 continue
             output.info(f"Checking remote: {remote.name}")
             try:
-                if self._legacy_update and need_update:
+                if self._legacy_update and need_update and reference.revision is None:
                     if not getattr(ConanProxy, "update_policy_legacy_warning", None):
                         ConanProxy.update_policy_legacy_warning = True
                         ConanOutput().warning("The 'core:update_policy' conf is deprecated and will "

--- a/test/integration/command/install/install_update_test.py
+++ b/test/integration/command/install/install_update_test.py
@@ -233,3 +233,15 @@ class TestUpdateOldPolicy:
         c.run("remove * -c")
         c.run("graph info --requires=pkg/0.1 --update -cc core:update_policy=legacy")
         assert f"pkg/0.1#{rev2} - Downloaded (r1)" in c.out
+
+    def test_lockfile(self):
+        # https://github.com/conan-io/conan/issues/18006
+        c = TestClient(default_server_user=True, light=True)  # needs server to fail
+        c.save_home({"global.conf": "core:update_policy=legacy"})
+        c.save({"dep/conanfile.py": GenConanfile("dep", "0.1"),
+                "pkg/conanfile.py": GenConanfile("pkg", "0.1").with_python_requires("dep/0.1")})
+        c.run("create dep")
+        c.run("lock create pkg --lockfile-out base.lock --build=* --update")
+        c.run("lock create pkg --lockfile base.lock --lockfile-out full.lock --build=* --update")
+        # it doesn't crash
+        assert "Generated lockfile" in c.out

